WooCommerce/Classes/ViewRelated/Hub Menu/HubMenuViewModel.swift

```diff
@@ -193,53 +212,57 @@ final class HubMenuViewModel: ObservableObject {
         settingsElements.append(Subscriptions())
     }
 
-    private func setupGeneralElements() {
-        generalElements = [Payments(iconBadge: shouldShowNewFeatureBadgeOnPayments ? .dot : nil),
-                           WoocommerceAdmin(),
-                           ViewStore(),
-                           Reviews()]
-        if generalAppSettings.betaFeatureEnabled(.inAppPurchases) {
-            generalElements.append(InAppPurchases())
-        }
-
-        inboxEligibilityChecker.isEligibleForInbox(siteID: siteID) { [weak self] isInboxMenuShown in
-            guard let self = self else { return }
-            if let index = self.generalElements.firstIndex(where: { item in
-                type(of: item).id == Reviews.id
-            }), isInboxMenuShown {
-                self.generalElements.insert(Inbox(), at: index + 1)
+    func setupGeneralElements() {
+        $shouldShowNewFeatureBadgeOnPayments
+            .combineLatest($isSiteEligibleForInbox,
+                           $isSiteEligibleForBlaze,
+                           $isSiteEligibleForGoogleAds)
+            .map { [weak self] combinedResult -> [HubMenuItem] in
+                let (shouldShowBadgeOnPayments, eligibleForInbox, eligibleForBlaze, eligibleForGoogleAds) = combinedResult
+                return self?.createGeneralElements(
+                    shouldShowBadgeOnPayments: shouldShowBadgeOnPayments,
+                    eligibleForGoogleAds: eligibleForGoogleAds,
+                    eligibleForBlaze: eligibleForBlaze,
+                    eligibleForInbox: eligibleForInbox
+                ) ?? []
```

Comment:
Super nit: I would prefer checking `self` in a guard and returning an empty array. This is because `createGeneralElements` always returns a non-nil value, and using a guard self pattern will make that clear.

Reply:
Done in fefad449778c57411d51601bd58498a5da48c333.